.github/workflows/docker.yml

```diff
@@ -0,0 +1,204 @@
+            github_project: ncsa/network_analysis
+          - name: preprocessing
+            dockerfile: Dockerfile
+            workdir: containerized_analytics/smile/preprocessing
+            hub_project: socialmediamacroscope/preprocessing
+            github_project: ncsa/preprocessing
+          - name: screen_name_prompt
+            dockerfile: Dockerfile
+            workdir: containerized_analytics/smile/screen_name_prompt
+            hub_project: socialmediamacroscope/screen_name_prompt
+            github_project: ncsa/screen_name_prompt
+          - name: sentiment_analysis
+            dockerfile: Dockerfile
+            workdir: containerized_analytics/smile/sentiment_analysis
+            hub_project: socialmediamacroscope/sentiment_analysis
+            github_project: ncsa/sentiment_analysis
+          - name: topic_modeling
+            dockerfile: Dockerfile
+            workdir: containerized_analytics/smile/topic_modeling
+            hub_project: socialmediamacroscope/topic_modeling
+            github_project: ncsa/topic_modeling
+    steps:
+    # checkout source code
+    - uses: actions/checkout@v2
+
+    # calculate version information
+    - name: version information
+      run: |
+        if [ "${{ github.event.release.target_commitish }}" != "" ]; then
+          BRANCH="${{ github.event.release.target_commitish }}"
+        elif [[ "${{github.event_name}}" == "pull_request" ]]; then
+            BRANCH="PR-${{github.event.pull_request.number}}"
+        else
+          BRANCH=${GITHUB_REF##*/}
+        fi
+        echo "GITHUB_BRANCH=${BRANCH}" >> $GITHUB_ENV
+
+        if [ "$BRANCH" == "main" ]; then
+          VERSION=$(cat manual_jb/content/_config.yml | grep "version" | head -1 | awk -F= "{print $2 }" | sed "s/[[:alpha:]|,|-|\"|:|#|[:space:]]//g" | sed "s/-//g")
+          VERSIONS="latest"
+          OLDVERSION=""
+          TMPVERSION=$VERSION
+          while [ "$OLDVERSION" != "$TMPVERSION" ]; do
+            VERSIONS="${VERSIONS} ${TMPVERSION}"
+            OLDVERSION="${TMPVERSION}"
+            TMPVERSION=$(echo ${OLDVERSION} | sed 's/\.[0-9]*$//')
+          done
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          echo "TAGS=${VERSIONS}" >> $GITHUB_ENV
+        elif [ "$BRANCH" == "develop" ]; then
+          echo "VERSION=develop" >> $GITHUB_ENV
+          echo "TAGS=develop" >> $GITHUB_ENV
+        else
+          echo "VERSION=testing" >> $GITHUB_ENV
+          echo "TAGS=${BRANCH}" >> $GITHUB_ENV
+        fi
+
+    - name: Print Version tag
+      run: echo "${{ env.TAGS }}"
+
+      # login to registries
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v2
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Build docker image
+      if: |
+        github.repository == env.MAIN_REPO &&
+        (github.event_name == env.GITHUB_BRANCH == 'main')
```

Comment:
what's github.event_name? I thought that's "pull_requests", "push" etc. Not sure it is actually the branch name?

Reply:
is this a typo, you might mean 
```
github.repository == env.MAIN_REPO && 
github.event_name == "push" && 
env.GITHUB_BRANCH == 'main'
```

Reply:
not type. right the one that you wrote.

Reply:
sorry im still a little bit confused by saying ` (github.event_name == env.GITHUB_BRANCH == 'main')` this will never be true? It's gonna be `"push" == "branch" == "main". Would you mind explain a little bit more :-(

Reply:
updated